Library/Formula/lcm.rb

```diff
@@ -0,0 +1,35 @@
+class Lcm < Formula
+  desc "libraries and tools for message passing and data marshalling"
+  homepage "https://lcm-proj.github.io/"
+  url "https://github.com/lcm-proj/lcm/releases/download/v1.2.1/lcm-1.2.1.zip"
+  sha256 "55b959409fc1c0cfa705209b679dc4b3043b68f14078e192b65c1b6021c37ea6"
+
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on :java => :recommended
+  depends_on "python3" => :optional
+
+  def install
+    # Make the directory for shared jars (if it doesn't already exist).
+    # This way, we should get:
+    #   /usr/local/share/java/lcm.jar -> /usr/local/Cellar/lcm/1.2.1/share/java/lcm.jar
+    # instead of:
+    #   /usr/local/share/java -> /usr/local/Cellar/lcm/1.2.1/share/java
+    # and hopefully avoid conflicts with any other forumlae that install
+    # shared jar files.
+    mkpath "/usr/local/share/java"
```

Comment:
Also, use `HOMEBREW_PREFIX` instead of `/usr/local`; they're not always the same.